src/plugins/auto_batch/src/compiled_model.cpp

```diff
@@ -39,63 +39,38 @@ CompiledModel::~CompiledModel() {
     auto workerRequestPtrAndId = GetWorkerInferRequest();
-    return std::make_shared<SyncInferRequest>(networkInputs,
-                                              networkOutputs,
-                                              workerRequestPtrAndId.first,
-                                              workerRequestPtrAndId.second,
-                                              m_device_info.batch_for_device,
-                                              m_batched_inputs,
-                                              m_batched_outputs);
+    auto async_infer_request = std::make_shared<ov::autobatch_plugin::SyncInferRequest>(
+        std::dynamic_pointer_cast<const ov::autobatch_plugin::CompiledModel>(shared_from_this()),
+        workerRequestPtrAndId.first,
+        workerRequestPtrAndId.second,
+        m_device_info.device_batch_size,
+        m_batched_inputs,
+        m_batched_outputs);
+    return async_infer_request;
 }
 
-InferenceEngine::IInferRequestInternal::Ptr CompiledModel::CreateInferRequestImpl(
-    const std::vector<std::shared_ptr<const ov::Node>>& inputs,
-    const std::vector<std::shared_ptr<const ov::Node>>& outputs) {
-    if (!this->_plugin || !_plugin->IsNewAPI())
-        return nullptr;
-    auto workerRequestPtrAndId = GetWorkerInferRequest();
-    return std::make_shared<SyncInferRequest>(inputs,
-                                              outputs,
-                                              workerRequestPtrAndId.first,
-                                              workerRequestPtrAndId.second,
-                                              m_device_info.batch_for_device,
-                                              m_batched_inputs,
-                                              m_batched_outputs);
-}
-
-std::pair<CompiledModel::WorkerInferRequest&, int> CompiledModel::GetWorkerInferRequest() {
+std::pair<std::shared_ptr<ov::autobatch_plugin::CompiledModel::WorkerInferRequest>, int>
+CompiledModel::GetWorkerInferRequest() const {
     auto num = m_num_requests_created++;
     std::lock_guard<std::mutex> lock(m_worker_requests_mutex);
-    auto batch_id = num % m_device_info.batch_for_device;
+    auto batch_id = num % m_device_info.device_batch_size;
     if (!batch_id) {  // need new request
         m_worker_requests.push_back(std::make_shared<WorkerInferRequest>());
         auto workerRequestPtr = m_worker_requests.back().get();
-        workerRequestPtr->_inferRequestBatched = {m_model_with_batch->CreateInferRequest(), m_model_with_batch._so};
-        workerRequestPtr->_batchSize = m_device_info.batch_for_device;
-        workerRequestPtr->_completionTasks.resize(workerRequestPtr->_batchSize);
-        workerRequestPtr->_inferRequestBatched->SetCallback(
+        workerRequestPtr->_infer_request_batched = {m_compiled_model_with_batch->create_infer_request(),
+                                                    m_compiled_model_with_batch._so};
```

Comment:
@ilyachur  I think we do not have meta plugin in auto batch, the proxy plugin is before auto batch plugin. What situation you mean for this?

Reply:
I mean that in theory we can get this situation, and we should think about it for the future, or just add a check in order to avoid overriding of low level so.

Reply:
ok, will put in next PR